worker/buildbot_worker/runprocess.py

```diff
@@ -611,7 +611,7 @@ def _spawnAsBatch(self, processProtocol, executable, args, env,
         twisted and cmd.exe with respect to escaping quotes"""
 
         tf = NamedTemporaryFile(mode='w+', dir='.', suffix=".bat",
-                                delete=False)
+                                delete=False, encoding=self.builder.unicode_encoding)
```

Comment:
I think you should also fix the tests to use BJK encoding when testing chineese chars

Reply:
The reason why test worker using __cp1252__ is that the codes below returns __cp1252__.
```python
if sys.platform.startswith("win"):
    def getpreferredencoding(do_setlocale=True):
        return _locale._getdefaultlocale()[1]
```

Reply:
Using BJK encoded bytes in unittest will not work because `bytes2NativeString` and `win32_batch_quote` will decode them using `self.builder.unicode_encoding` which may not be BJK encoding.